sdk/src/main/java/co/omise/android/ui/ExpiryDateEditText.kt

```diff
@@ -73,6 +75,15 @@ class ExpiryDateEditText : OmiseEditText {
         override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
             if (s == null || s.length > MAX_CHARS) return
 
+            // Define a regular expression pattern to match non-numeric characters and DATE_SEPARATOR
+            val pattern: Pattern = Pattern.compile("[^0-9" + Pattern.quote(DATE_SEPARATOR) + "]")
+
+            // Use a Matcher to find any non-numeric characters in the string
+            val matcher: Matcher = pattern.matcher(s)
+            if(matcher.find()){
+                return
+            }
```

Comment:
Can we move this to a function? Maybe to a different class if this is useful in other parts of the code.

```
fun findAProperFunName(s: CharSequence?) {
    // Define a regular expression pattern to match non-numeric characters and DATE_SEPARATOR
    val pattern: Pattern = Pattern.compile("[^0-9" + Pattern.quote(DATE_SEPARATOR) + "]")

    // Use a Matcher to find any non-numeric characters in the string
    val matcher: Matcher = pattern.matcher(s)
    return matcher.find()
}

override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
    if (s == null || s.length > MAX_CHARS) return

    if (findAProperFunName(s)) {
        return
    }

    // rest of the code
}
```

Reply:
or maybe combine top two `if` statements.

```
    if (s == null || s.length > MAX_CHARS || findAProperFunName(s)) {
        return
    }
```

Reply:
> Can we move this to a function? Maybe to a different class if this is useful in other parts of the code.
> 
> ```
> fun findAProperFunName(s: CharSequence?) {
>     // Define a regular expression pattern to match non-numeric characters and DATE_SEPARATOR
>     val pattern: Pattern = Pattern.compile("[^0-9" + Pattern.quote(DATE_SEPARATOR) + "]")
> 
>     // Use a Matcher to find any non-numeric characters in the string
>     val matcher: Matcher = pattern.matcher(s)
>     return matcher.find()
> }
> 
> override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
>     if (s == null || s.length > MAX_CHARS) return
> 
>     if (findAProperFunName(s)) {
>         return
>     }
> 
>     // rest of the code
> }
> ```

@aashishgurung  I would say to keep it as inline code directly as this is the pattern followed in the sdk currently and each text field validation is incapsulated in the class of the textfield itself (Maybe create a function in the class itslef). So date validations are within the date textfield class. Card validations are within the card text field class and so on. We do not have a centralized validation utility class. It's a nice to have I would say as doing that now will lead to a large refactor. Also these validations are kind of not totally reusable as Date validation is just for date validation and not anywhere else outside of the date class....